api.py

```diff
@@ -0,0 +1,277 @@
+            '/crawls',
+            view_func=self.list_crawls,
+            methods=['GET'],
+        )
+        self.add_url_rule(
+            '/crawls',
+            view_func=self.stop,
+            methods=['DELETE'],
+        )
+        self.add_url_rule(
+            '/',
+            view_func=self.home,
+            methods=['GET'],
+        )
+
+    def home(self):
+        routes = [
+            {
+                "url": "/spiders",
+                "method": "GET"
+            },
+            {
+                "url": "/spiders",
+                "method": "POST",
+                "arguments": {
+                    "spider": "name of the spider to run"
+                }
+            },
+            {
+                "url": "/spiders/:spider_id",
+                "method": "DELETE",
+                "arguments": {
+                    "spider_id": "uuid of the spider to close"
+                }
+            },
+            {
+                "url": "/crawls",
+                "method": "GET"
+            },
+            {
+                "url": "/crawls",
+                "method": "DELETE"
+            },
+            {
+                "url": "/database",
+                "method": "GET"
+            },
+            {
+                "url": "/database",
+                "method": "POST",
+                "arguments": {
+                    "file": "json file containing the database dump"
+                }
+            },
+            {
+                "url": "/database",
+                "method": "DELETE"
+            }
+        ]
+        result = self._get_meta_response({"routes": routes})
+        return jsonify(result), 200
+
+    def list_spiders(self):
+        spiders = self.process.spider_loader.list()
+        return jsonify({"spiders": spiders, "status": "success"}), 200
+
+    def run_spider(self):
+        post_data = request.get_json()
+        spider = post_data.get('spider')
+        if spider == 'who_iris':
+            spider = who_iris_spider.WhoIrisSpider()
+        elif spider == 'nice':
+            spider = nice_spider.NiceSpider()
+        else:
+            return '', 400
+        spider_id = str(uuid.uuid4())
+        self.process.crawl(spider, uuid=spider_id)
+        crawl = self.process.join()
+        self.database.insert_spider(spider.name, spider_id)
+        crawl.addBoth(self.on_success)
+        return jsonify({
+            "data": {
+              "status": "running",
+              "spider": spider.name,
+              "_id": spider_id
+            }
+        }), 200
+
+    def on_success(self, data):
+        self.database._close_all_spiders()
+
+    def close_spider(self, spider_id):
+        for crawl in self.process.crawlers:
+            if crawl.spider.uuid == uuid:
+                crawl.stop()
+                return jsonify({
+                    "data":
+                        {"status": "success", "_id": spider_id}
+                    }), 200
+        return '', 400
```

Comment:
and this code should probably be something in the 5xx space i think. https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

Again up to you